lightning/src/routing/network_graph.rs

```diff
@@ -277,11 +328,11 @@ where C::Target: chain::Access, L::Target: Logger
 		let mut result = Vec::with_capacity(batch_amount as usize);
 		let nodes = self.network_graph.nodes.read().unwrap();
 		let mut iter = if let Some(pubkey) = starting_point {
-				let mut iter = nodes.range((*pubkey)..);
+				let mut iter = nodes.range(NodeId(pubkey.serialize())..);
```

Comment:
Instead of calling `serialize` everywhere we need to do a conversion, I'd prefer if we define a constructor as:

```rust
pub fn from_pubkey(pubkey: &PublicKey) -> Self {
	NodeId(pubkey.serialize())
}
```

And then the field could be private with a slice conversion function when needed outside of the module (e.g. in `router.rs`).

```rust
pub fn as_slice(&self) -> &[u8] {
	&self.0
}
```